scrapers/scrapers/spiders/mo-bills.py

```diff
@@ -0,0 +1,687 @@
+    else:
+        return Chamber.LOWER
+
+
+def get_action(actor, action):
+    # Alright. This covers both chambers and everything else.
+    categories = []
+    for flag, acat in flags:
+        if flag in action:
+            categories.append(acat)
+
+    return categories or None
+
+
+class BillsSpider(Spider):
+    name = "mo-bills"
+    session = None
+    chamber = None
+
+    subjects = defaultdict(list)
+
+    def __init__(self, session=None, chamber=None, **kwargs):
+        self.session = session
+        self.chamber = chamber
+        super().__init__(**kwargs)
+
+    def start_requests(self):
+        self.parse_subjects(self.session)
+
+        if self.chamber in [Chamber.UPPER, None]:
+            # scrape senate bills
+            sb_url = 'https://www.senate.mo.gov/23info/BTS_Web/BillList.aspx?SessionType=R'
+            yield Request(url=sb_url, callback=self.parse_upper_chamber, headers=custom_headers)
+
+        if self.chamber in [Chamber.LOWER, None]:
+            # scrape house bills
+            hb_xml_url = 'https://documents.house.mo.gov/xml/SessionListTest.xml'
+            yield Request(url=hb_xml_url, callback=self.parse_lower_chamber, headers=custom_headers)
+
+    def parse_upper_chamber(self, response):
+        for bill_table in response.xpath("//a[contains(@id, 'hlBillNum')]"):
+            yield Request(url=response.urljoin(bill_table.xpath('@href').get()), callback=self.parse_senate_billpage)
+
+    def parse_lower_chamber(self, response):
+        first_id = None
+        for session in response.xpath('//Session'):
+            session_id = session.xpath('./ID/text()').get()
+            if not first_id:
+                first_id = session_id
+
+            session_year = session.xpath('./SessionYear/text()').get()
+            session_code = session.xpath('./SessionCode/text()').get()
+            session_code = '' if session_code == 'R' else session_code
+            if f'{session_year}{session_code}' == self.session:
+                break
+        if first_id == session_id:
+            # extract data from https://documents.house.mo.gov/xml/BillListTest.xml
+            bill_list_url = 'https://documents.house.mo.gov/xml/BillListTest.xml'
+            yield Request(url=bill_list_url, callback=self.parse_house_bill_list)
+        else:
+            # upzip https://documents.house.mo.gov/xml/{ID}.zip and extract data from xmls
+            zipped_xml_url = f'https://documents.house.mo.gov/xml/{session_id}.zip'
+            zip_response = requests.get(zipped_xml_url, headers=custom_headers)
+            zip_file = ZipFile(BytesIO(zip_response.content))
+            # read 221-BillList.xml in zip file
+            bill_list_content = zip_file.read(
+                f'{session_id}/{session_id}-BillList.xml')
+            bl_response = Selector(text=bill_list_content, type='xml')
+
+            for bill in bl_response.xpath('//BillXML'):
+                bill_url = bill.xpath('./BillXMLLink/text()').get()
```

Comment:
This section (173-178) looks like it is repeated below in `parse_house_bill_list()` (lines 193-198). Could `parse_house_bill_list()` be used fore both the "first" (unzipped) session and for an older (zipped) session? If so I think that would be preferable. It is a common mistake when maintaining scrapers to accidentally fix a selector/string in one section, and not realize it is repeated somewhere else.

Reply:
@jessemortenson I've updated this issue.